File: app/src/main/java/cn/zjl/datacollector/data/entity/SurveyLineEntity.java
package cn.zjl.datacollector.data.entity;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

/**
 * 测线实体类
 */
@Entity(tableName = "survey_lines")
public class SurveyLineEntity {
    @PrimaryKey(autoGenerate = true)
    public long id;
    
    // 所属工程 ID
    public long projectId;
    
    // 测线名称/编号
    public String name;
    
    // 测线描述
    public String description;
    
    // 创建时间
    public long createdAt;
    
    // 最后修改时间
    public long updatedAt;
}

File: app/src/main/java/cn/zjl/datacollector/data/entity/SurveyLineEntity.kt
package cn.zjl.datacollector.data.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "data_line")
data class SurveyLineEntity(
    @PrimaryKey(autoGenerate = true) var id: Long = 0,
    var name: Float = 0f,
    var type: Int = 0,
    var use: Int = 0,
    var note: String = "",
    var projectId: Long = 0,
    var createdAt: Long = 0,
    var updatedAt: Long = 0
)
